jbpm-runtime-manager/src/main/java/org/jbpm/runtime/manager/impl/jpa/ExecutionErrorInfo.java

```diff
@@ -103,6 +105,15 @@ public Long getActivityId() {
         return super.getActivityId();
     }
 
+    @Override
+    public void setErrorMessage(String errorMessage) {
+        String trimmedErrorMessage = errorMessage;
+        if (trimmedErrorMessage != null && trimmedErrorMessage.length() > ERROR_LOG_LENGTH) {
+            trimmedErrorMessage = trimmedErrorMessage.substring(0, ERROR_LOG_LENGTH);
+        }
+        super.setErrorMessage(trimmedErrorMessage);
```

Comment:
Perhaps we may use _Apache Commons Lang_ **abbreviate** method, adding 3 dots at the end, to indicate that it's trimmed. Notice that this method would throw _IllegalArgumentException_ if length is equal or less than 3 (should not be a common length to use). What do you think?

```suggestion
        super.setErrorMessage(StringUtils.abbreviate(errorMessage, ERROR_LOG_LENGTH));
```

Reply:
Assuming the exception is not equivalent. So In spite is good catch I would stick to this implementation rather than throw a potential runtime exception. (we would trade off one exception for another even if it is unlikely to happen)